conan/tools/scm/git.py

```diff
@@ -84,6 +84,19 @@ def clone(self, url, target="", args=None):
         self._conanfile.output.info("Cloning git repo")
         self.run('clone "{}" {} {}'.format(url, " ".join(args), target))
 
+    def fetch_commit(self, url, commit):
+        """
+        Experimental: does a 1 commit fetch and checkout, instead of a full clone,
+        should be faster.
+        """
+        if os.path.exists(url):
+            url = url.replace("\\", "/")  # Windows local directory
+        self._conanfile.output.info("Shallow fetch of git repo")
+        self.run('init')
```

Comment:
For anyone worried about this init as I was, reading https://git-scm.com/docs/git-init#_description says that calling `init` multiple times is ok

Reply:
In any case, this keeps failing in Linux because of https://github.com/conan-io/conan/issues/13081#issuecomment-1425940875, no idea why